src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -374,6 +376,59 @@ public void sortFriendList(Comparator<Friend> comparator) {
         observableDirectory.setObservableDirectory(directory);
     }
 
+    /**
+     * Checks if accommodation object date is within the model's travel plan start and end date.
+     * @return false if accommodation start date and end date is within travel plan start date and end date.
+     */
+    public boolean isValidAccommodationDate(WanderlustDate startDate, WanderlustDate endDate) {
+        WanderlustDate travelPlanStartDate = directory.getStartDate();
+        WanderlustDate travelPlanEndDate = directory.getEndDate();
+
+        if (startDate.getValue().isBefore(travelPlanStartDate.getValue())
+                || startDate.getValue().isAfter(travelPlanEndDate.getValue())) {
+            return false;
+        }
+
+        if (endDate.getValue().isBefore(travelPlanStartDate.getValue())
```

Comment:
I think it's bad "SLAP" here. Perhaps can just do startDate.isBefore(WanderlustDate)

Reply:
fixed